.github/workflows/pre-commit.yml

```
@@ -0,0 +1,17 @@
+
+on:
+  workflow_dispatch:
+#  pull_request:
+#  push:
+#    branches: [main]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)